### PR TITLE
New control file for CCN policy

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

This PR introduces an initial control file for CCN-STIC-610A22.
It is a Benchmark developed by National Cryptological Center of Spain, in alignment to the National Security Schema (ENS)

This control file is not used in any profile at this moment.
The intention is to already bring this initial version to the project so we can collaborate to improve it.

#### Rationale:

New control file for CCN-STIC-610A22.

#### Review Hints:

The `pending` requirements are those which demands more collaboration mainly about the interpretation of the requirement. Sometimes it is not clear the intention of the requirement or multiple interpretations are possible.
It would be great to have more context from people involved with CCN-STIC-610A22 development or use.

All the `planned` requirements were interpreted and some candidate rules were selected.
For these requirements a review of the rules is necessary to move them from `related_rules` to `rules`.

### CODE REVIEW

File: controls/ccn_rhel9.yml [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+      - intermediate
+      - advanced
+    status: planned
+    related_rules:
+      - file_permissions_audit_configuration
+      - file_ownership_audit_configuration
+      - file_groupownership_audit_configuration
+
+  - id: A.3.SEC-RHEL5
+    title: It is implemented a detailed audit based on subcategories
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:95]:
```suggestion
    title: A detailed audit has been implemented based on subcategories
```

File: controls/ccn_rhel9.yml [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+    levels:
+      - advanced
+    status: pending
+    notes: |-
+      The intention here is probably to audit changes in /etc/pam.d files, but we need to confirm
+      this assumption and get more context.
+
+  #[A.4] Manipulación de los ficheros de configuración
+  - id: A.4.SEC-RHEL1
+    title: Common users do not have local administrator permissions or are included in a sudo group
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:203]:
```suggestion
    title: Common users do not have local administrator permissions and are not included in a sudo group
```
This phrasing may be more clear.

File: controls/ccn_rhel9.yml [hunk 3]
```diff
@@ -0,0 +1,836 @@
...
+      - basic
+      - intermediate
+      - advanced
+    status: pending
+    notes: |-
+      Related to nosuid, noexec and nodev options but in /boot. More context is needed.
+
+  #[A.5] Suplantación de la identidad.
+  - id: A.5.SEC-RHEL1
+    title: Control login and impersonation permissions
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:241]:
```suggestion
    title: Login and impersonation permissions are controlled
```
Modified it for an increased consistency with the rest of the titles.

File: controls/ccn_rhel9.yml [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+    levels:
+      - intermediate
+      - advanced
+    status: planned
+    related_rules:
+      - sudo_add_use_pty
+      - use_pam_wheel_for_su
+
+  - id: A.5.SEC-RHEL2
+    title: Elevation attempts are controlled by defining sudoers users and groups
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:252]:
```suggestion
    title: Elevation attempts are controlled by defining users and sudoer groups
```
NIT

File: controls/ccn_rhel9.yml [hunk 5]
```diff
@@ -0,0 +1,836 @@
...
+      - basic
+      - intermediate
+      - advanced
+    status: planned
+    related_rules:
+      - configure_crypto_policy
+      - var_system_crypto_policy=default_policy
+
+  - id: A.5.SEC-RHEL5
+    title: Recurring password change required
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:287]:
```suggestion
    title: Recurring password change is required
```
NIT

File: controls/ccn_rhel9.yml [hunk 6]
```diff
@@ -0,0 +1,836 @@
...
+      - accounts_password_set_min_life_existing
+      - accounts_maximum_age_login_defs
+      - var_accounts_maximum_age_login_defs=45
+      - accounts_password_set_max_life_existing
+      - accounts_password_warn_age_login_defs
+      - var_accounts_password_warn_age_login_defs=10
+      - accounts_password_set_warn_age_existing
+
+  - id: A.5.SEC-RHEL6
+    title: Secure protocols are used for network authentication processes
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:306]:
```suggestion
    title: Secure protocols are used for the network authentication processes
```
NIT

File: controls/ccn_rhel9.yml [hunk 7]
```diff
@@ -0,0 +1,836 @@
...
+    levels:
+      - basic
+      - intermediate
+      - advanced
+    status: planned
+    related_rules:
+      - configure_ssh_crypto_policy
+
+  - id: A.5.SEC-RHEL7
+    title: Control network session inactivity
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:317]:
```suggestion
    title:Network session inactivity is controlled
```
Modified it for an increased consistency with the rest of the titles.

File: controls/ccn_rhel9.yml [hunk 8]
```diff
@@ -0,0 +1,836 @@
...
+      - advanced
+    status: planned
+    related_rules:
+      - sshd_idle_timeout_value=15_minutes
+      - sshd_set_idle_timeout
+      - sshd_set_keepalive
+      - var_sshd_set_keepalive=0
+
+  - id: A.5.SEC-RHEL8
+    title: Control local and remote console inactivity
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:331]:
```suggestion
    title: Local and remote console inactivity is controlled
```
Modified it for an increased consistency with the rest of the titles.

File: controls/ccn_rhel9.yml [hunk 9]
```diff
@@ -0,0 +1,836 @@
...
+      - accounts_password_pam_minclass
+      - accounts_password_pam_minlen
+      - accounts_password_pam_retry
+      - var_password_pam_minclass=4
+      - var_password_pam_minlen=14
+      - ensure_root_password_configured
+      - no_empty_passwords_etc_shadow
+
+  - id: A.11.SEC-RHEL4
+    title: During login, the system displays a text in compliance with the organization's standards or policies
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:603]:
```suggestion
    title: During login, the system displays a text in compliance with the organization's standards or directives
```

File: controls/ccn_rhel9.yml [hunk 10]
```diff
@@ -0,0 +1,836 @@
...
+      - sshd_enable_warning_banner_net
+      - banner_etc_motd
+      - motd_banner_text=cis_banners
+      - banner_etc_issue
+      - login_banner_text=cis_banners
+      - banner_etc_issue_net
+      - remote_login_banner_text=cis_banners
+
+  - id: A.11.SEC-RHEL5
+    title: Control network access to the system
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:623]:
```suggestion
    title: Network access to the system is controlled
```

File: controls/ccn_rhel9.yml [hunk 11]
```diff
@@ -0,0 +1,836 @@
...
+      - advanced
+    status: planned
+    related_rules:
+      - dconf_gnome_disable_automount
+      - dconf_gnome_disable_automount_open
+      - dconf_gnome_disable_autorun
+
+  #[A.15] Modificación de la información.
+  - id: A.15.SEC-RHEL1
+    title: Control the use of removable storage media
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:711]:
```suggestion
    title: The use of removable storage media is controlled
```

File: controls/ccn_rhel9.yml [hunk 12]
```diff
@@ -0,0 +1,836 @@
...
+    status: planned
+    related_rules:
+      - set_password_hashing_algorithm_systemauth
+      - set_password_hashing_algorithm_passwordauth
+      - set_password_hashing_algorithm_logindefs
+      - var_password_hashing_algorithm=SHA512
+
+  #[A.23] Manipulación del hardware.
+  - id: A.23.SEC-RHEL1
+    title: Control the installation and use of any device connected to the equipment
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:759]:
```suggestion
    title: The installation and use of any device connected to the equipment is controlled
```

File: controls/ccn_rhel9.yml [hunk 13]
```diff
@@ -0,0 +1,836 @@
...
+      - intermediate
+      - advanced
+    status: planned
+    related_rules:
+      - package_usbguard_installed
+      - service_usbguard_enabled
+      - usbguard_generate_policy
+
+  - id: A.23.SEC-RHEL2
+    title: Restrict dynamic mounting and unmounting of file systems
```
Comment (cortesana) [inline controls/ccn_rhel9.yml:772]:
```suggestion
    title: The dynamic mounting and unmounting of file systems is restricted
```

**Review by cortesana — COMMENTED** [review]
Small changes here and there, more related to the consistency of the title format rather than the translations. Thanks for your work.

File: controls/ccn_rhel9.yml [hunk 14]
```diff
@@ -0,0 +1,836 @@
...
+      - intermediate
+      - advanced
+    notes: |-
+      The policy doesn't have any section where this would fit better.
+    status: automated
+    rules:
+      - var_authselect_profile=sssd
+      - enable_authselect
+
+  #[A.3] Manipulación de los registros de actividad (log).
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply controls/ccn_rhel9.yml:42]: I think we can remove them. Thanks to bringing this.

**Comment by marcusburghardt** [comment]
Once this PR is merged, we can start working on its `planned` items.

File: controls/ccn_rhel9.yml [hunk 15]
```diff
@@ -0,0 +1,824 @@
...
+      - kernel_module_udf_disabled
+
+  - id: A.8.SEC-RHEL5
+    title: Application execution is controlled
+    original_title: Se controla la ejecución de aplicaciones.
+    levels:
+      - advanced
+    status: pending
+    notes: |-
+      This might be related to SELinux.
```
Comment (Mab879) [inline controls/ccn_rhel9.yml:474]: Or fapolicyd for application control.

File: docs/manual/developer/03_creating_content.md [hunk 16]
```diff
@@ -983,20 +983,21 @@ This is a complete schema of the YAML file format.
 ```
 id: policy ID (required key)
 title: short title (required key)
+original_title: used as a reference for policies not yet available in english
```
Comment (Mab879) [inline docs/manual/developer/03_creating_content.md:986]:
```suggestion
original_title: used as a reference for policies not yet available in English
```

File: controls/ccn_rhel9.yml [hunk 17]
```diff
@@ -0,0 +1,824 @@
...
+  - id: intermediate
+    inherits_from:
+      - basic
+  - id: advanced
+    inherits_from:
+      - intermediate
+
+controls:
+  - id: reload_dconf_db
+    title: Reload Dconf database
```
Comment (Mab879) [inline controls/ccn_rhel9.yml:18]: Title should be Title case per the [style guide](https://complianceascode.readthedocs.io/en/latest/manual/developer/04_style_guide.html#controls).

**Review by Mab879 — CHANGES_REQUESTED** [review]

File: controls/ccn_rhel9.yml (hunk 15 above)
Comment (marcusburghardt) [reply controls/ccn_rhel9.yml:474]: Good. We have to clarify these `pending` requirements at some point.

Comment (marcusburghardt) [reply controls/ccn_rhel9.yml]: Included

**Review by Mab879 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**